packages/vaadin-grid/test/templates.test.js

```diff
@@ -442,47 +425,74 @@ describe('slotted templates', () => {
   });
 
   ['header', 'footer'].forEach((container) => {
-    it(`should change the ${container} template`, () => {
+    it(`should change the ${container} template`, async () => {
       const newTemplate = document.createElement('template');
       newTemplate.classList.add(container);
       newTemplate.setAttribute('slot', `grid-column-${container}-template`);
       newTemplate.innerHTML = `${container}-bar`;
 
       wrapper.removeChild(wrapper.querySelector(`template.${container}`));
       wrapper.appendChild(newTemplate);
+      await nextRender();
       flushGrid(grid);
-      const column = newTemplate.assignedSlot.parentElement;
-      column._templateObserver.flush();
 
       expect(getContainerCellContent(grid.$[container], 0, 5).textContent).to.equal(`${container}-bar`);
     });
   });
 
-  it(`should change the body template`, () => {
+  it(`should change the body template`, async () => {
     const newTemplate = document.createElement('template');
     newTemplate.setAttribute('slot', `grid-column-template`);
     newTemplate.innerHTML = `bar-[[index]]`;
 
     slotted.removeChild(slotted.querySelector(`template`));
     slotted.appendChild(newTemplate);
+    await nextRender();
     flushGrid(grid);
-    const column = newTemplate.assignedSlot.assignedSlot.parentElement;
-    column._templateObserver.flush();
 
     expect(getContainerCellContent(grid.$.items, 0, 5).textContent).to.equal('bar-0');
   });
 });
 
-describe('observed', () => {
-  let observed;
-
-  beforeEach(() => {
-    observed = fixtureSync('<observed-grid></observed-grid>');
-    flushGrid(observed.$.grid);
-  });
-
-  it('should invoke once', async () => {
-    await aTimeout(300); // Filter's debounce time is 200ms
-    expect(observed.count).to.equal(1);
-  });
-});
+// class ObservedGrid extends PolymerElement {
+//   static get template() {
+//     return html`
+//       <vaadin-grid items='["foo"]' id="grid">
+//         <template class="row-details"> [[index]] </template>
+//         <vaadin-grid-column>
+//           <template class="header">
+//             <vaadin-grid-filter path="index"></vaadin-grid-filter>
+//           </template>
+//           <template>[[index]] [[_count(item, index)]]</template>
+//         </vaadin-grid-column>
+//       </vaadin-grid>
+//     `;
+//   }
+
+//   static get properties() {
+//     return {
+//       count: Number
+//     };
+//   }
+
+//   _count() {
+//     this.count = this.count || 0;
+//     this.count++;
+//   }
+// }
+//
+// customElements.define('observed-grid', ObservedGrid);
+//
+// describe('observed', () => {
+//   let observed;
+
+//   beforeEach(() => {
+//     observed = fixtureSync('<observed-grid></observed-grid>');
+//     flushGrid(observed.$.grid);
+//   });
+
+//   it('should invoke once', async () => {
```

Comment:
The body renderer is invoked at least 4 times when initializing a cell (probably it can be optimized later).

Since we force the `item` property to re-render in the template renderer on each renderer call (https://github.com/vaadin/web-components/pull/2067), the following test no longer passes.

**Expected:** 1, **Actual:** 4.